src/dmd/dtemplate.d

```diff
@@ -6125,55 +6125,80 @@ extern (C++) class TemplateInstance : ScopeDsymbol
-            enum foo = helper();
-        }
-        ---
-
-        since it was the easiest starting point of implementation but it can and
-        should be expanded more later.
     */
     final bool isDiscardable()
     {
-        if (aliasdecl is null)
+        if (aliasdecl is null) // not eponymous
             return false;
 
-        auto v = aliasdecl.isVarDeclaration();
-        if (v is null)
-            return false;
+        if (auto v = aliasdecl.isVarDeclaration())
+        {
+            /** Detect & discard:
+                ---
+                template foo(args...) {
+                    some_basic_type_or_string helper() { .... }
+                    enum foo = helper();
+                }
+                ---
+            */
 
-        if (!(v.storage_class & STC.manifest))
-            return false;
+            if (!(v.storage_class & STC.manifest))
+                return false;
 
-        // Currently only doing basic types here because it is the easiest proof-of-concept
-        // implementation with minimal risk of side effects, but it could likely be
-        // expanded to any type that already exists outside this particular instance.
-        if (!(v.type.equals(Type.tstring) || (v.type.isTypeBasic() !is null)))
-            return false;
+            // Currently only doing basic types here because it is the easiest proof-of-concept
+            // implementation with minimal risk of side effects, but it could likely be
+            // expanded to any type that already exists outside this particular instance.
+            if (!(v.type.equals(Type.tstring) || v.type.isTypeBasic()))
+                return false;
 
-        // Static ctors and dtors, even in an eponymous enum template, are still run,
-        // so if any of them are in here, we'd better not assume it is trivial lest
-        // we break useful code
-        foreach(member; *members)
+            // Static ctors and dtors, even in an eponymous enum template, are still run,
+            // so if any of them are in here, we'd better not assume it is trivial lest
+            // we break useful code
+            foreach (member; *members)
+            {
+                if (member.hasStaticCtorOrDtor() ||
+                    member.isStaticDtorDeclaration() ||
+                    member.isStaticCtorDeclaration())
+                {
+                    return false;
+                }
+            }
+        }
+        else if (auto a = aliasdecl.isAliasDeclaration())
         {
-            if(member.hasStaticCtorOrDtor())
-                return false;
-            if(member.isStaticDtorDeclaration())
+            /** Detect & discard:
+                ---
+                template foo(args...) {
+                    import bar : Symbol;
+                    alias foo = Symbol!args;
+                }
+                ---
+            */
+
+            // Only *type* aliases
+            if (a.aliassym)
```

Comment:
This doesn't just trigger on type aliases. a.aliassym can be symbol, like a function symbol.